contentgrid-spring-data-rest/src/test/java/com/contentgrid/spring/test/fixture/invoicing/InvoicingApplicationTests.java

```diff
@@ -138,15 +150,28 @@ void doInTransaction(ThrowingCallable callable) {
         });
     }
 
+    String formatInstant(Instant date) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss z", Locale.ENGLISH)
+                .withZone(ZoneId.of("GMT"));
+        return formatter.format(date);
+    }
+
+    public static void loginUser(String username) {
+        var user = new User(username, "password", Set.of());
+        var authentication = new UsernamePasswordAuthenticationToken(user, user.getPassword(), user.getAuthorities());
+        var context = SecurityContextHolder.getContext();
+        context.setAuthentication(authentication);
+    }
```

Comment:
can you explain how this is related to `@WithMockUser` on the class-level ?

Reply:
`@WithMockUser` sets the `SecurityContext` to contain an authenticated `User` with the given username. This `loginUser()` overrides `SecurityContext.authentication` to contain a new authenticated `User`. So that following requests are send by the new user. This is used to test whether `lastModifiedBy` can be altered when a different user makes a PUT request for example.

Reply:
so the "problem" is that if you override `@WithMockUser("other")` on the test-method level, the before-each also runs with user `"other"` ?

Reply:
With `@WithMockUser` alone, you can only test one user per test:
If a test is marked with `@WithMockUser("John")` and it calls a function that modifies an object in a repository with `@WithMockUser("Bob")`, `lastModifiedDate` of that object will be updated, but `lastModifiedBy` will contain `"John"`.

Reply:
and something like this does not work: https://docs.spring.io/spring-security/reference/servlet/test/mockmvc/authentication.html#test-mockmvc-securitycontextholder-rpp ?